JetAnalyzers/bin/jet_response_fitter_x.cc

```diff
@@ -255,6 +271,584 @@ int main(int argc,char**argv)
+  testfit->SetParameter(0, hrsp->GetBinContent(binmax));
+  testfit->SetParameter(1, xbinmax);
+  testfit->FixParameter(1, xbinmax);
+  testfit->SetParameter(2, 0.10);
+  testfit->SetParLimits(2, 0., 1.e+1); 
+  gausLeft = new TF1("gausLeft", gaus.data(), 0., 2.);
+  do {
+    hrsp->Fit(testfit, "Q0", "", minGaus, xbinmax); // Q: quiet mode; 0: do not draw default canvas
+
+    TH1F * hrspXaxis = hXaxis(hrsp, minGaus, xbinmax); // set axis for KS
+    TF1 *fit=hrsp->GetFunction("testfit");
+    fit->SetNpx(hrspXaxis->GetNbinsX());
+    ks=hrspXaxis->KolmogorovTest((TH1D*)fit->GetHistogram()); 
+    chi2 = testfit->GetChisquare()/max(1, testfit->GetNDF()); 
+
+    --bindown;
+    minGaus = hrsp->GetXaxis()->GetBinLowEdge(bindown);
+    if ( chi2 < 5. && ks>0.05) {
+      gausLeft->SetParameter(0, testfit->GetParameter(0));
+      gausLeft->SetParameter(1, testfit->GetParameter(1));
+      gausLeft->SetParameter(2, testfit->GetParameter(2));	  
+    }
+  } while ( chi2 < 5. && minGaus > 0. && ks>0.05 );
+  // sometimes chi2 may exceed 5 in 1st interation->set defauld value
+  if(gausLeft->GetParameter(0)==0) gausLeft->SetParameter(0, testfit->GetParameter(0));
+  if(gausLeft->GetParameter(1)==0) gausLeft->SetParameter(1, testfit->GetParameter(1));
+  if(gausLeft->GetParameter(2)==0) gausLeft->SetParameter(2, testfit->GetParameter(2));
+
+  ++bindown;
+  minGaus = hrsp->GetXaxis()->GetBinCenter(bindown);
+
+ 
+  //////////////////////////////////////////////////////////////////////////
+  // left side expo pre-fit
+  //////////////////////////////////////////////////////////////////////////
+  cout<<"\nProcessing fit expo left..."<< endl;
+  double minExpLeft = hrsp->GetXaxis()->GetBinLowEdge(bindown - 3); // min bin of the expo (R1)
+  expLeft = new TF1("expLeft", expo.data(), 0., 2.);
+  if ( minGaus > 0 && hrsp->Integral(hrsp->FindBin(0), hrsp->FindBin(minGaus)) > 0. ){
+    delete testfit;
+    testfit = new TF1("testfit", expo.data(), 0., 2.);
+    testfit->SetParameter(0, gausLeft->Eval(minGaus));
+    testfit->SetParameter(1, 0.1);
+    testfit->SetParameter(2, minGaus);
+    testfit->FixParameter(2, minGaus);
+    do {
+      hrsp->Fit(testfit, "Q0", "", minExpLeft, minGaus);
+      --bindown;
+      minExpLeft = hrsp->GetXaxis()->GetBinCenter(bindown);
+      chi2 = testfit->GetChisquare()/max(1, testfit->GetNDF()); 
+      if ( chi2 < 5. ) {
+	expLeft->SetParameter(0, testfit->GetParameter(0));
+	expLeft->SetParameter(1, testfit->GetParameter(1));
+	expLeft->SetParameter(2, testfit->GetParameter(2));
+      }
+    }
+    while ( chi2 < 5. && minExpLeft > 0 );
+    // sometimes chi2 may exceed 5 in 1st interation->se t default value
+    if(expLeft->GetParameter(0)==0) expLeft->SetParameter(0, testfit->GetParameter(0));
+    if(expLeft->GetParameter(1)==0) expLeft->SetParameter(1, testfit->GetParameter(1));
+    if(expLeft->GetParameter(2)==0) expLeft->SetParameter(2, testfit->GetParameter(2));
+
```

Comment:
This procedure seems very similar to the Gaussian fit procedure. Is there any way to generalize it and move it to a new function? Sorry if this is a stupid question.